crates/krator/examples/moose.rs

```diff
@@ -331,6 +362,70 @@ impl Operator for MooseTracker {
             }),
         }
     }
+
+    #[cfg(feature = "admission-webhook")]
+    async fn admission_hook_tls(&self) -> anyhow::Result<krator::admission::AdmissionTls> {
+        let client = self.shared.read().await.client.clone();
+        let secret_name = Moose::admission_webhook_secret_name();
+
+        let secret = kube::Api::<Secret>::namespaced(client, "default")
+            .get(&secret_name)
+            .await?;
+
+        Ok(admission::AdmissionTls::from(&secret)?)
+    }
+}
+
+async fn apply_crd(client: &kube::Client) -> anyhow::Result<CustomResourceDefinition> {
+    info!("installing moose crd");
+    let mut crd = Moose::crd();
+    let crd_name = crd.metadata.name.as_ref().unwrap();
+
+    let api = kube::Api::<CustomResourceDefinition>::all(client.to_owned());
+
+    let created_crd;
+    if let Ok(existing_crd) = api.get(crd_name).await {
+        crd.metadata.resource_version = Some(existing_crd.resource_ver().unwrap());
+        created_crd = api.replace(&crd_name, &PostParams::default(), &crd).await?;
+    } else {
+        created_crd = api.create(&PostParams::default(), &crd).await?;
+    }
+
+    Ok(created_crd)
+}
+
+#[cfg(feature = "admission-webhook")]
+async fn setup_cluster(client: &kube::Client) -> anyhow::Result<()> {
```

Comment:
I think this would ideally be run automatically by `krator` when a given feature flag is enabled.

Reply:
I don't think I really agree here ... there are many different way of installing CRD and the other resources, especially due to different RBAC permissions. The function itself is kind of trivial when used like that and as it ships within the example, users can copy it if they like the approach.

Reply:
resolving this as this would mean pulling the operations, that need a lot of permissions, into krator